File: code/src/climate_analogs_jl/climate_analogs.jl
module ClimateAnalogs

# Function to load all the *fn.jl files
function load_functions()
    # Get the path to the directory containing the *fn.jl files
    script_dir = joinpath(dirname(@__FILE__), "..", "climate_analogs_jl")
    fn_files = filter(x -> endswith(x, r"fns?\.jl"), readdir(script_dir))
    
    # Load each *fn.jl file
    for file in fn_files
        include(joinpath(script_dir, file))
    end
end

# Load all the functions when the module is imported
load_functions()

export all_functions

end # module

File: code/src/climate_analogs_R/climate_analogs.R
# Function to load all the *fn.jl files
load_functions <- function() {
    # Get the path to the directory containing the *fn.jl files
    script_dir <- file.path(getwd(), "code/src/climate_analogs_R")
    fn_files <- list.files(script_dir, pattern = "fns?\\.R$", full.names = TRUE)
    fn_files <- c(fn_files, file.path(script_dir, "as.data.table.R"))

    # Load each *fn.jl file
    for (file in fn_files) {
        source(file)
    }
}

# Load all the functions when the module is imported
load_functions()

# Export all the functions
all_functions <- ls()
